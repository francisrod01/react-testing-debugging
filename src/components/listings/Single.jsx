import React from 'react';

const Single = ({ item }) => (
  <li className="col l6 s12">
    <div className="card">
      <div className="card-image">
        <img src={item.image} alt={item.title} />
      </div>
      <div className="card-content">
        <span className="card-title">{item.title}</span>
        <p>{item.description}</p>
        <div className="row">
          <div className="col l6 center">
            <blockquote><h5>Released: {item.release_date}</h5></blockquote>
          </div>
          <div className="col l6 center">
            <blockquote><h5>Views: {item.views}</h5></blockquote>
          </div>
        </div>
      </div>
      <div className="card-action">
        <a href={item.link} target="_blank" rel="noopener noreferrer">Link to course</a>
      </div>
    </div>
    <br />
  </li>
);

export default Single;
