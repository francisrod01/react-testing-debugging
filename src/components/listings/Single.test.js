import React from 'react';
import renderer from 'react-test-renderer';

import Single from './Single';


const item = {
  "id": 1,
  "title": "Building an App with ReactJS and MeteorJS",
  "category": "react",
  "description": "Meteor and React are a powerhouse combination. Meteor gives you a fast, easy-to-use solution for data management across clients and servers, and React gives you a way to structure your app's UI from reusable components. The combination allows you to create your dream apps: dynamic, data driven, and cross-platform.",
  "link": "https://www.lynda.com/React-js-tutorials/Building-App-React-js-MeteorJS/533228-2.html",
  "release_date": "12/21/2016",
  "views": "31,266",
  "image": "https://cdn.lynda.com/course/533228/533228-636179098122239600-16x9.jpg"
};

describe('Single snapshots', () => {
  it('Single snapshot test', () => {
    const component = renderer.create(<Single item={item} />);
    const tree = component.toJSON();
    
    expect(tree).toMatchSnapshot();
  });
});
