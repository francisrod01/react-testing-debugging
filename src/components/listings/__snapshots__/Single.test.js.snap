exports[`Single snapshots Single snapshot test 1`] = `
<li
  className="col l6 s12">
  <div
    className="card">
    <div
      className="card-image">
      <img
        alt="Building an App with ReactJS and MeteorJS"
        src="https://cdn.lynda.com/course/533228/533228-636179098122239600-16x9.jpg" />
    </div>
    <div
      className="card-content">
      <span
        className="card-title">
        Building an App with ReactJS and MeteorJS
      </span>
      <p>
        Meteor and React are a powerhouse combination. Meteor gives you a fast, easy-to-use solution for data management across clients and servers, and React gives you a way to structure your app\'s UI from reusable components. The combination allows you to create your dream apps: dynamic, data driven, and cross-platform.
      </p>
      <div
        className="row">
        <div
          className="col l6 center">
          <blockquote>
            <h5>
              Released: 
              12/21/2016
            </h5>
          </blockquote>
        </div>
        <div
          className="col l6 center">
          <blockquote>
            <h5>
              Views: 
              31,266
            </h5>
          </blockquote>
        </div>
      </div>
    </div>
    <div
      className="card-action">
      <a
        href="https://www.lynda.com/React-js-tutorials/Building-App-React-js-MeteorJS/533228-2.html"
        target="_blank">
        Link to course
      </a>
    </div>
  </div>
  <br />
</li>
`;
